cli/azd/cmd/root.go

```diff
@@ -150,38 +148,21 @@ func BuildCmd[F any](
 
 		ctx = tools.WithInstalledCheckCache(ctx)
 
-		actionResult, err := action.Run(ctx)
+		middleware.Use(middleware.Build(flags, opts, actionOptions, console, initDebugMiddleware))
+		middleware.Use(middleware.Build(flags, opts, actionOptions, console, initTelemetryMiddleware))
+		middleware.Use(middleware.Build(flags, opts, actionOptions, console, initCommandHooksMiddleware))
```

Comment:
What if we instead just created each middleware inline by calling the relevant `init` functions? I think it makes it clearer what each middleware component depends on.

Reply:
Updated to use IoC container.